bundles/com.espressif.idf.core/src/com/espressif/idf/core/resources/ResourceChangeListener.java

```diff
@@ -128,6 +138,33 @@ private void updateLaunchBar(IResource resource, int kind, int flags) throws Cor
 		}
 	}
 
+	private void renameProjectInCmakeList(IResource resource)
+	{
+		IProject project = (IProject) resource;
+		Path cMakeListLocation = new File(project.getLocation() + "/CMakeLists.txt").toPath(); //$NON-NLS-1$
+		List<String> fileContent;
+		try
+		{
+			fileContent = new ArrayList<>(Files.readAllLines(cMakeListLocation));
+			for (int i = 0; i < fileContent.size(); i++)
+			{
+
+				Pattern p = Pattern.compile("(project[(].+[)])"); //$NON-NLS-1$
```

Comment:
`"(project[(].+[)])"`

This can be extracted to a constant

Reply:
done